### PR TITLE
Update rule.md

### PR DESCRIPTION
**Pull request by tomek-i**
As per my email, I think this rule needs to be updated. The original example is using an "invalid" value for `Sum`, however 0 is a valid value: Consider the following examples:
```
Sum(0,0) ;
Sum(int.MaxValue,1);
Sum(-1,1);
```
For this mathematical operation any integer is a valid value.

Apart from that, the `Sum` method is not (correctly implemented) and it is valid case use and throw a `NotImplementedException" to mark this functionality as "still in development". The same goes for a Test, it clearly shows the current state as "still in development". 

The MSDN uses `NotImplementedException` in their examples: https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-mstest

If we want to give examples of TDD https://www.geeksforgeeks.org/test-driven-development-tdd/ then, the examples and steps should follow strictly TDD and the approach of writing minimal passing test eg:
```cs
[Test]
public void ShouldAddTwoNumbers()
{
   var calculator = new Calculator();
   var result = calculator.Sum(10, 11);
   Assert.Equal(21, result);
}

//implementation example in calculator
public int Sum(int x, int y)
{
   return 21;
}
```
then, you would write a second test (or additionally use [Values(...)] (https://docs.nunit.org/articles/nunit/writing-tests/attributes/values.html) or [Random(...)] (https://docs.nunit.org/articles/nunit/writing-tests/attributes/values.html), attribute to run a few more eg,  edge cases and see that the 2nd test failed and then follow the motto of TDD: red, green, refactor

### CODE REVIEW

**Comment by wicksipedia** [comment]
Did you want to mention that you can look for tests that give false positives through mutation testing? e.g. https://stryker-mutator.io/docs/stryker-net/introduction/

File: rules/make-sure-that-the-test-can-be-failed/rule.md [hunk 1]
```diff
@@ -42,25 +42,61 @@ public int Sum(int x, int y)
...
 // The method to test in class Calculator ...
+public int IsPositive(int x)
+{
+   //NOTE: ths method has a clear "invalid" value
+   return -1;
+}
 public int Sum(int x, int y)
 {
-   return 0;
+   //NOTE: this method does not have a clear "invalid" value and throws a NotImplementedException and inclused a TODO marker
```
Comment (wicksipedia) [inline rules/make-sure-that-the-test-can-be-failed/rule.md:91]:
```suggestion
   //NOTE: this method does not have a clear "invalid" value and throws a NotImplementedException and includes a TODO marker
```

**Review by wicksipedia — CHANGES_REQUESTED** [review]
> You should make sure that the unit tests you create can actually fail. A test that never fails is not useful to anyone.

The double negative in the opening is really throwing me off. Can you fix it up as well?

I was thinking `It's important that the unit tests you develop are capable of failing. A test that can never fail isn't helpful for anyone.`

**Comment by tomek-i** [comment]
> > You should make sure that the unit tests you create can actually fail. A test that never fails is not useful to anyone.
> 
> The double negative in the opening is really throwing me off. Can you fix it up as well?
> 
> I was thinking `It's important that the unit tests you develop are capable of failing. A test that can never fail isn't helpful for anyone.`

Very nice! I made another addition to it adding that the user has seen the test fail too.

**Review by wicksipedia — APPROVED** [review]

File: rules/make-sure-that-the-test-can-be-failed/rule.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ redirects:
 
 ---
 
-You should make sure that the unit tests you create can actually fail. A test that never fails is not useful to anyone.
+It's important that the unit tests you develop are capable of failing and that you have seen it fail. A test that can never fail isn't helpful for anyone.
```
Comment (danielmackay) [inline rules/make-sure-that-the-test-can-be-failed/rule.md:19]:
```suggestion
It's important that the unit tests you develop are capable of failing now or in future.  A test that can never EVER fail isn't helpful for anyone.
```

**Comment by drwharris** [comment]
We should always look for our test sweet to PASS. Therefore a test should never plan to fail. It should test failure and make sure they are handled in an expected way.

Currently this reads like you should write tests that fail.

**Comment by tomek-i** [comment]
Thanks for the feedback @drwharris 
In true TDD fashion you should write the test first which is failing because there is no code that statisfies the test. Then you write the minimum code to make the test pass. Refactor. The code is now immutable if you want to change the code then you need to write another test that proofs the code needs change to statisfy the test / all tests.

But in General you are correct, there should not be a failed test in master. Do you want to paraphrase or mark th sections that you think need to be updated?

**Comment by SamBWagner** [comment]
> Thanks for the feedback @drwharris In true TDD fashion you should write the test first which is failing because there is no code that statisfies the test. Then you write the minimum code to make the test pass. Refactor. The code is now immutable if you want to change the code then you need to write another test that proofs the code needs change to statisfy the test / all tests.
> 
> But in General you are correct, there should not be a failed test in master. Do you want to paraphrase or mark th sections that you think need to be updated?

This of course assumes that you're doing TDD. Which is seldom the case. At least in my experience. Great changes though :) approved from me

**Review by UlyssesMaclaren — APPROVED** [review]
